x-pack/plugins/cloud/public/plugin.test.ts

```diff
@@ -212,6 +219,101 @@ describe('Cloud Plugin', () => {
+
+      beforeEach(() => {
+        consoleMock = jest.spyOn(console, 'debug').mockImplementation(() => {});
+      });
+
+      afterEach(() => {
+        consoleMock.mockRestore();
+      });
+
+      const setupPlugin = async ({
+        config = {},
+        securityEnabled = true,
+        currentUserProps = {},
+        isCloudEnabled = true,
+        failHttp = false,
+      }: {
+        config?: Partial<CloudConfigType>;
+        securityEnabled?: boolean;
+        currentUserProps?: Record<string, any>;
+        isCloudEnabled?: boolean;
+        failHttp?: boolean;
+      }) => {
+        const initContext = coreMock.createPluginInitializerContext({
+          id: isCloudEnabled ? 'cloud-id' : null,
+          base_url: 'https://cloud.elastic.co',
+          deployment_url: '/abc123',
+          profile_url: '/profile/alice',
+          organization_url: '/org/myOrg',
+          full_story: {
+            enabled: false,
+          },
+          chat: {
+            enabled: false,
+          },
+          ...config,
+        });
+
+        const plugin = new CloudPlugin(initContext);
+
+        const coreSetup = coreMock.createSetup();
+        const coreStart = coreMock.createStart();
+
+        if (failHttp) {
+          coreSetup.http.get.mockImplementation(() => {
+            throw new Error('HTTP request failed');
+          });
+        }
+
+        coreSetup.getStartServices.mockResolvedValue([coreStart, {}, undefined]);
+
+        const securitySetup = securityMock.createSetup();
+        securitySetup.authc.getCurrentUser.mockResolvedValue(
+          securityMock.createMockAuthenticatedUser(currentUserProps)
+        );
+
+        const setup = plugin.setup(coreSetup, securityEnabled ? { security: securitySetup } : {});
+
+        return { initContext, plugin, setup, coreSetup };
+      };
+
+      it('chatConfig is not retrieved if cloud is not enabled', async () => {
+        const { coreSetup } = await setupPlugin({ isCloudEnabled: false });
+        expect(coreSetup.http.get).not.toHaveBeenCalled();
+      });
+
+      it('chatConfig is not retrieved if security is not enabled', async () => {
+        const { coreSetup } = await setupPlugin({ securityEnabled: false });
+        expect(coreSetup.http.get).not.toHaveBeenCalled();
+      });
+
+      it('chatConfig is not retrieved if chat is enabled but url is not provided', async () => {
+        // @ts-expect-error 2741
+        const { coreSetup } = await setupPlugin({ config: { chat: { enabled: true } } });
+        expect(coreSetup.http.get).not.toHaveBeenCalled();
+      });
+
+      it('chatConfig is not retrieved if internal API fails', async () => {
+        const { coreSetup } = await setupPlugin({
+          config: { chat: { enabled: true, chatURL: 'http://chat.elastic.co' } },
+          failHttp: true,
+        });
+        expect(coreSetup.http.get).toHaveBeenCalled();
+        expect(consoleMock).toHaveBeenCalled();
+      });
```

Comment:
are these the right assertions on this test? do you mean `.not.toHaveBeenCalled()`?

Reply:
The failHttp flag means that the test calls for HTTP, but the response is bad, so the error is logged to the console.